libs/angular/src/lib/slider/slider.component.ts

```diff
@@ -0,0 +1,88 @@
+import { ChangeDetectionStrategy } from '@angular/compiler'
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core'
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms'
+import {
+  getSliderTrackBackground,
+  randomId,
+  sliderColors,
+} from '@sebgroup/extract'
+
+interface SliderStyle {
+  background?: string
+}
+
+@Component({
+  selector: 'ngg-slider',
+  templateUrl: './slider.component.html',
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: NggSliderComponent,
+      multi: true,
+    },
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class NggSliderComponent implements ControlValueAccessor, OnInit {
+  private _value = 0
+  @Input() name = `${randomId}-slider`
+  @Input() min = 0
+  @Input() max = 100
+  @Input() step = 1
+  @Input() errorMessage = ''
+  @Input() hasTextbox = false
+  @Input() disabled = false
+  @Input() set value(val: number) {
+    this._value = val ?? 0
+    this.handleChange()
+  }
+  get value(): number {
+    return this._value
+  }
+  @Output() sliderChange = new EventEmitter<number>()
+  @Output() sliderTouch = new EventEmitter<boolean>()
+  onChangeFn?: (val: number) => void
+  onTouchedFn?: VoidFunction
+  style: SliderStyle = {}
+
+  ngOnInit(): void {
+    this.setTrackBackground()
+  }
+
+  onFocus(): void {
+    this.sliderTouch.emit(true)
+    this.onTouchedFn && this.onTouchedFn()
```

Comment:
The [abstract control](https://angular.io/api/forms/AbstractControl) interface notes that `A control is marked touched once the user has triggered a blur event on it`